### PR TITLE
Updating load-themed-styles to the latest tools and the updated typescript options.

### PR DESCRIPTION
**Pull request by iclanton**
This change also puts the "console.warn" behind a "DEBUG" variable to make the console less chatty.

### CODE REVIEW

File: src/tests/index.test.ts [hunk 1]
```diff
@@ -6,7 +8,7 @@ import {
   loadStyles,
   configureLoadStyles,
   IThemingInstruction
-} from './index';
+} from './../index';
```
Comment (dzearing) [inline src/tests/index.test.ts]: this seems broken.

File: src/index.ts [hunk 2]
```diff
@@ -159,36 +158,30 @@ export function detokenize(styles: string): string {
...
+  // Then join the array to produce the final CSS string.
+  const resolvedArray: (string | undefined)[] = (splitStyleArray || []).map((currentValue: IThemingInstruction) => {
+    const themeSlot: string | undefined = currentValue.theme;
+    if (themeSlot) {
+      // A theming annotation. Resolve it.
+      const themedValue: string | undefined = theme ? theme[themeSlot] : undefined;
+      const defaultValue: string = currentValue.defaultValue || 'inherit';
+
+      // Warn to console if we hit an unthemed value even when themes are provided, unless "DEBUG" is false or
+      //  isn't present.
```
Comment (dzearing) [inline src/index.ts]: This should read "unless DEBUG global variable (in the global scope) is explicitly set to false".

If it's undefined, you're still logging, right?

File: src/tests/index.test.ts (hunk 1 above)
Comment (iclanton) [inline src/tests/index.test.ts]: I moved the tests.ts file into a "tests" directory.

---
In reply to: [115051501](https://github.com/Microsoft/load-themed-styles/pull/7#discussion_r115051501) [](ancestors = 115051501)

File: src/index.ts (hunk 2 above)
Comment (iclanton) [inline src/index.ts]: The way it's written if it's truthy or not defined (i.e. - "DEBUG" doesn't exist at all), it'll log, but if it's undefined it won't log.

Basically - if someone isn't using webpack (or another bundler that uses the "DEBUG" variable), they'll get logs. If they're using webpack and DEBUG=true, they'll get logs, but if DEBUG=false or DEBUG=undefined, they won't

---
In reply to: [115052022](https://github.com/Microsoft/load-themed-styles/pull/7#discussion_r115052022) [](ancestors = 115052022)

File: src/tests/index.test.ts (hunk 1 above)
Comment (dzearing) [inline src/tests/index.test.ts]: oh ok could you just change it to `../index` then? :) no need for a ./ in the front

File: src/index.ts (hunk 2 above)
Comment (dzearing) [inline src/index.ts]: Right I mean the comment on line 171 doesn't match what you just sais. 

"Warn to console ..., unless "DEBUG" ... isn't present."

should read:

"Warn to console  ..., unless "DEBUG is false"

Comment (iclanton) [inline src/index.ts]: I can do that.

---
In reply to: [115062599](https://github.com/Microsoft/load-themed-styles/pull/7#discussion_r115062599) [](ancestors = 115062599)

**Pull request merged**